sbncode/CAFMaker/FillReco.cxx

```diff
@@ -384,19 +384,66 @@ namespace caf
     }
   }
 
-  void FillTrackPlaneCalo(const anab::Calorimetry &calo, float constant, caf::SRTrackCalo &srcalo) {
+  void FillTrackPlaneCalo(const anab::Calorimetry &calo, 
+        const std::vector<art::Ptr<recob::Hit>> &hits,
+        bool fill_calo_points, float fillhit_rrstart, float fillhit_rrend, 
+        const detinfo::DetectorPropertiesData &dprop,
+        caf::SRTrackCalo &srcalo) {
+
+    // Collect info from Calorimetry
     const std::vector<float> &dqdx = calo.dQdx();
     const std::vector<float> &dedx = calo.dEdx();
     const std::vector<float> &pitch = calo.TrkPitchVec();
+    const std::vector<float> &rr = calo.ResidualRange();
+    const std::vector<geo::Point_t> &xyz = calo.XYZ();
+    const std::vector<size_t> &tps = calo.TpIndices();
+
     srcalo.charge = 0.;
     srcalo.ke = 0.;
     srcalo.nhit = 0;
+
+    float rrmax = !rr.empty() ? *std::max_element(rr.begin(), rr.end()) : 0.;
+
     for (unsigned i = 0; i < dedx.size(); i++) {
+      // Save the points we need to
+      if (fill_calo_points && (
+          (rrmax - rr[i]) < fillhit_rrstart || // near start
+          rr[i] < fillhit_rrend)) { // near end
+
+        // Point information
+        caf::SRCaloPoint p;
+        p.rr = rr[i];
+        p.dqdx = dqdx[i];
+        p.dedx = dedx[i];
+        p.pitch = pitch[i];
+        p.t = dprop.ConvertXToTicks(xyz[i].x(), calo.PlaneID());
+
+        // lookup the wire -- the Calorimery object makes this
+        // __way__ harder than it should be
+        for (const art::Ptr<recob::Hit> &h: hits) {
+          if (h.key() == tps[i]) {
+            p.wire = h->WireID().Wire;
+            p.sumadc = h->SummedADC();
+            p.integral = h->Integral();
+          }
```

Comment:
Should this search always succeed? What if it fails? We get the default-initialized values? Do we want to assert instead?